File: 0061-rotate-list/0061-rotate-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* rotateRight(ListNode* head, int k) {
        if(head==nullptr||head->next==nullptr||k==0)
        {
            return head;
        }
        int size=1;
        ListNode *temp=head;
        while(temp->next!=nullptr)
        {
            size++;
            temp=temp->next;
        }
        k=k%size;
        for(int i = 0; i < k; i++){
            ListNode *temp = head;

            while(temp->next->next != NULL){
                temp = temp->next;
            }

            temp->next->next = head;
            head = temp->next;
            temp->next = NULL;
        }

        return head;
    }
};

File: 0061-rotate-list/0061-rotate-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode rotateRight(ListNode head, int k) {
        ListNode dummy = new ListNode(-1);
        dummy.next = head;
        ListNode temp = head;
        int c = 0;
        while(temp!=null){
            c++;
            temp = temp.next;
        }
        if(c==0){
            return null;
        }
        k=k%c;
        while(k>0){
        ListNode prev = null;
        ListNode curr=head;
            while(curr.next!=null){
                prev=curr;
                curr=curr.next;
            }
            curr.next=head;
            head=curr;
            prev.next=null;
            k--;
        }
        dummy.next=head;
        return dummy.next;
    }
}